meteor/lib/Rundown.ts

```diff
@@ -122,24 +122,23 @@ export function getResolvedSegment (showStyleBase: ShowStyleBase, rundown: Rundo
 
 	// fetch all the parts for the segment
 	let partsE: Array<PartExtended> = []
-	const parts = segment.getParts()
 
-	if (parts.length > 0) {
+	const segmentsAndParts = rundown.getSegmentsAndPartsSync()
+	const parts = segmentsAndParts.parts
+	// let segments = segmentsAndParts.segments
+	const partsInSegment = _.filter(parts, p => p.segmentId === segment._id)
```

Comment:
Like this: this is essentially an unnecessary pass though the array, at least from the UIs point of view. This could have been filtered at the query stage.